apps/api/src/app/workflows/usecases/create-workflow-override/create-workflow-override.usecase.ts

```diff
@@ -0,0 +1,85 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+
+import {
+  NotificationTemplateEntity,
+  NotificationTemplateRepository,
+  TenantEntity,
+  TenantRepository,
+  WorkflowOverrideRepository,
+} from '@novu/dal';
+
+import { CreateWorkflowOverrideCommand } from './create-workflow-override.command';
+import { CreateWorkflowOverrideResponseDto } from '../../dto/create-workflow-override-response.dto';
+
+@Injectable()
+export class CreateWorkflowOverride {
+  constructor(
+    private tenantRepository: TenantRepository,
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private workflowOverrideRepository: WorkflowOverrideRepository
+  ) {}
+
+  async execute(command: CreateWorkflowOverrideCommand): Promise<CreateWorkflowOverrideResponseDto> {
+    const { tenant, workflow } = await this.extractEntities(command);
+
+    return await this.workflowOverrideRepository.create({
+      _environmentId: command.environmentId,
+      _tenantId: tenant._id,
+      _workflowId: workflow._id,
+      active: command.active,
+      preferenceSettings: command.preferenceSettings,
+    });
+  }
+
+  private async extractEntities(
+    command: CreateWorkflowOverrideCommand
+  ): Promise<{ tenant: TenantEntity; workflow: NotificationTemplateEntity }> {
+    const tenant = await this.tenantRepository.findOne({
+      _environmentId: command.environmentId,
+      identifier: command.tenantIdentifier,
+    });
+
+    if (!tenant) {
+      throw new NotFoundException(`Tenant with identifier ${command.tenantIdentifier} is not found`);
+    }
+
+    let workflow: NotificationTemplateEntity | null = null;
+
+    if (!command.triggerIdentifier && !command._workflowId) {
+      throw new BadRequestException(`Either triggerIdentifier or _workflowId must be provided`);
+    }
+
+    if (command.triggerIdentifier) {
+      workflow = await this.notificationTemplateRepository.findByTriggerIdentifier(
+        command.environmentId,
+        command.triggerIdentifier
+      );
+
+      if (!workflow) {
+        throw new NotFoundException(`Workflow with identifier ${command.triggerIdentifier} is not found`);
+      }
+
+      return { tenant, workflow };
+    }
+
+    if (command._workflowId) {
+      workflow = await this.notificationTemplateRepository.findOne({
+        _environmentId: command.environmentId,
+        _id: command._workflowId,
+      });
+
+      if (!workflow) {
+        throw new NotFoundException(`Workflow with _workflowId ${command._workflowId} is not found`);
+      }
```

Comment:
and this, the code will be more clean :)

Reply:
yes, but the error message won't be informative as before, but i think i am ok with it :)